### PR TITLE
Main character rotation

### LINKED ISSUE #1135: [FEATURE] New character interactions in Main Screen
## 🚀 Feature
This task involves incorporating two new interactions for the character within the main screen:
1. Character Model Rotation: Enable players to rotate the character by swiping in the desired direction. After the player halts rotation, the model should revert to its initial position after a set duration (initially 3 seconds).
2. Display Character Information: Implement functionality to showcase character details upon tapping the character. This action will trigger the presentation of the character's information scene, enhancing user engagement and accessibility to character-specific details.

### PR DESCRIPTION
**Pull request by AngieDutra**
Closes #1135 

## Motivation

The character in main screen does not rotate on drag and it would look more fancy if it did.

## Summary of changes

This PR adds a rotation to the character model in main screen and character info
It should:
- rotate on drag and stop when the finger is released
- after 3 seconds it should rotate to it's original rotation position
  - in addition, it should rotate to the direction closer to the original position (right or left)
- the transition should be as smooth as possible (keeping in mind a refactor is coming)
- in muflus's case the rotation should ignore the model's size, since it includes the weapon, and rotate as if the anchor was at the middle of the character (excluding the weapon)
- ATENTION: there is a little bit of input lag TDL in a future PR

## Checklist
- [x] I have tested the changes locally.
- [x] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [x] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [x] Tested in Android.

### CODE REVIEW

File: client/Assets/Scripts/UI/RotateUIModel.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        if (touch.phase == TouchPhase.Moved)
+        {
+            rotationX = Quaternion.Euler(0f, -touch.deltaPosition.x * rotationSpeed, 0f);
+            model.rotation *= rotationX;
+        }
+    }
+
+    public void OnPointerDown(PointerEventData eventData)
+    {
+        StopCoroutine(coroutine);
```
Comment (tkz00) [inline client/Assets/Scripts/UI/RotateUIModel.cs:44]: I've found adding this conditional fixes the issue.

```
public void OnPointerDown(PointerEventData eventData)
{
    if(coroutine != null)
    {
        StopCoroutine(coroutine);
    }
}
```

**Review by tkz00 — CHANGES_REQUESTED** [review]
It looks and feels absolutely amazing, I love it.

The only thing is that when I try to rotate the character I get an null reference exception, it doesn't affect the functionality but should still be taken care of.

image

File: client/Assets/Scripts/UI/RotateUIModel.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+public class RotateUIModel : MonoBehaviour, IDragHandler, IPointerUpHandler, IPointerDownHandler
+{
+    [SerializeField]
+    float rotationSpeed = 1f;
+
+    [SerializeField]
+    GameObject modelContainer;
+    Transform model;
+    Touch touch;
+    Vector2 touchPosition;
```
Comment (tkz00) [inline client/Assets/Scripts/UI/RotateUIModel.cs:15]: The "touchPosition", "resetRotation" and "rotationValue" variables are unused.

File: client/Assets/Scripts/UI/RotateUIModel.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    void Awake()
+    {
+        StartCoroutine(GetModel());
+    }
+
+    IEnumerator GetModel()
+    {
+        yield return new WaitUntil(() => modelContainer.GetComponentInChildren<Animator>() != null);
+        model = modelContainer.GetComponentInChildren<Animator>().transform.parent.transform;
```
Comment (BertovDev) [inline client/Assets/Scripts/UI/RotateUIModel.cs:29]: I believe this is more readable 
`model = modelContainer.transform.GetChild(0);`

File: client/Assets/Textures/ModelsTexture/playerModelMainScreen.renderTexture
Comment (BertovDev) [inline client/Assets/Textures/ModelsTexture/playerModelMainScreen.renderTexture:1]: We are using the same RenderTexture in `characterInfo` so this is causing the characters to look totally flat.
Captura de pantalla 2023-11-27 a la(s) 11 19 59

**Review by BertovDev — CHANGES_REQUESTED** [review]
Stunning! just a few comments:

- I believe `CharacterInfo` should have the same behavior for the character models.

File: client/Assets/Scripts/UI/RotateUIModel.cs (hunk 2 above)
Comment (AngieDutra) [reply client/Assets/Scripts/UI/RotateUIModel.cs]: Done ✨

File: client/Assets/Scripts/UI/RotateUIModel.cs (hunk 3 above)
Comment (AngieDutra) [reply client/Assets/Scripts/UI/RotateUIModel.cs]: true, I changed it ✨

File: client/Assets/Textures/ModelsTexture/playerModelMainScreen.renderTexture
Comment (AngieDutra) [reply client/Assets/Textures/ModelsTexture/playerModelMainScreen.renderTexture:1]: Fixed it! ✨ And also added the rotation feature here too

**Review by tkz00 — APPROVED** [review]
Great, looks amazing!

**Review by BertovDev — APPROVED** [review]
Stunning!

**Pull request merged**